analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MultilineBlocksWithoutBrace.cs

```diff
@@ -1,173 +1,228 @@
+        else Act.First(); Act.Second();
+        //   ^^^^^^^^^^^^ Secondary
+        //                ^^^^^^^^^^^^^ @-1
+    }
 
-            if (a)
-            while (b)
-               Console.WriteLine();
+    void ElseIf(bool condition)
+    {
+        if (condition) Act.Other();
+        else if (condition) Act.First(); Act.Second();
+        //                  ^^^^^^^^^^^^ Secondary
+        //                               ^^^^^^^^^^^^^ @-1
+    }
+}
 
-            Console.WriteLine();
+class FirstOneSameLineAndSecondWithIndentation
+{
+    void While(bool condition)
+    {
+        while (condition) Act.First(); // Secondary
+            Act.Second(); // Noncompliant
+    }
 
-            while (a)
-            if (b)
-               Console.WriteLine();
+    void ForEach(string[] args)
+    {
+        foreach (var arg in args) Act.First(arg); // Secondary
+            Act.Second(); // Noncompliant
+    }
 
-            Console.WriteLine();
-        }
+    void If(bool condition)
+    {
+        if (condition) Act.First(); // Secondary
+            Act.Second(); // Noncompliant
+    }
+
+    void Else(bool condition)
+    {
+        if (condition) Act.Other();
+        else Act.First(); // Secondary
+            Act.Second(); // Noncompliant
+    }
+
+    void ElseIf(bool condition)
+    {
+        if (condition) Act.Other();
+        else if (condition) Act.First(); // Secondary
+            Act.Second(); // Noncompliant
+    }
+}
+
+class Nested
+{
+    void While(bool condition)
+    {
+        while(true)
+            while (condition) Act.First(); // Secondary
+                Act.Second(); // Noncompliant
+    }
+
+    void ForEach(string[] args)
+    {
+        foreach(var _ in args)
+            foreach (var arg in args) Act.First(arg); // Secondary
+                Act.Second(); // Noncompliant
+    }
 
-        void TestWeirdAlignment(bool a)
-        {
-            try {
-              if (a)
-                Console.WriteLine(); // This statement is aligned with the '{' of the try on purpose to fix https://github.com/SonarSource/sonar-dotnet/issues/264
-            } finally { }
+    void If(bool condition)
+    {
+        if (condition)
+            if (condition) Act.First(); // Secondary
+                Act.Second(); // Noncompliant
+    }
+}
+
+class Other
+{
+    int WithReturnStatement(bool condition)
+    {
+        while (condition)
+            Act.First();
+        //  ^^^^^^^^^^^^ Secondary
+            return Act.Second();
+        //  ^^^^^^^^^^^^^^^^^^^^
+    }
+
+    void ZeroIdentation(bool condition)
+    {
+        if (condition)
+Act.First(); // Secondary
+Act.Second(); // Noncompliant
+    }
+
+    void AlternativeIndentation(bool condition)
+    {
+        try {
+          if (condition)
+            Act.First(); // This statement is aligned with the '{' of the try on purpose to fix https://github.com/SonarSource/sonar-dotnet/issues/264
         }
+        finally { }
     }
 }
+
```

Comment:
if(condition) ; // <-- note the empty statement
    Act.Second();
Let's add some cases with empty statement too.
```
if(condition) ; ;
if(condition) ; // <-- note the empty statement
    Act.Second();
if(condition) ; Act.Second();

if(condition) Act.First; ; // <-- note the empty statement
if(condition) Act.First; 
    ; // <-- note the empty statement
```

Reply:
Please add these cases with emptystatements

Reply:
Added them. Some of the noncompliant statements are not correct (yet). I struggled a bit with the Secondary declarations.

Reply:
I'm not saying we **have to** report correctly on these. We can also decide to accept it as FN, or intentionally explicitly ignore them.
Because "Do not use empty statements" sounds like a different rule

Reply:
Ignoring them sounds like a good idea. I would assume there already should be a rule dealing with empty statements?

Reply:
S1116